syllabus/1-Cryptography/zk-proofs/slides.md

```diff
@@ -0,0 +1,446 @@
+---
+
+## What can we show?
+
+- NP relation: `function(statement, witness) -> bool`
+
+- Prover knows a witness for a statement:
+
+  - They want to show that they know it (_a proof of knowledge_)
+
+  - ... Without revealing anything about the witness (_ZK_)
+
+---
+
+## ZK Proof Interface
+
+- NP relation: `function(statement, witness) -> bool`
+
+- `prove(statement, witness) -> proof`
+
+- `verify(statement, proof) -> bool`
+
+---
+
+## Examples
+---
+
+## ZK Proof Example
+
+_Example:_ Schnorr signatures are ZK Proofs
+- $(s, e)$ such that $s = (\textrm{Random blinding factor }r) - \\ \textrm{Private Key} \times e$
+- They show that the prover knows the private key (the discrete log of the public key) without revealing anything about it.
+- The statement is the public key and the witness the private key.
+
+---
+
+## The ZK Proof Example we study in this module
+
+_Example:_ Prover knows a non-trivial factorization of N.
+
+- Prover wants to prove that they know $N = r \times s$  without revealing $r$ or $s$.
+
+- The Prover should convince us they know the two non-trivial integer $r$ and $s$ that is $r,s \neq 1$ such that:
+
+ $r\times s = N$ 
+
+- Without revealing any other information about $r$ or $s$.
+
+- The statement is $N$ and the witnesses are $r$ and $s$.
+
+
+---
+## ZK Proof properties.
+ - Completeness: If the claim is true, then it *must* pass `verify(statement, proof) == true`
+ - Statistical Knowledge Soundness: If the prover does not know the witness then `verify(statement, proof) == false` with high probability.
+ - Zero Knowledge: the proof reveals nothing about the witness that was not revealed by the statement itself.
+ - The common way of implementing zero knowledge protocol is by means of zk-SNARK.
+
+Notes:
+ 
+ - We prove Zero knowldege by showing a party without any knowledge of the witness, could fool an external party who was not been part of the interaction that they interacted with the prover and got a valid proof form the prover. 
+---
+
+## zk-SNARKs
+
+---
+
+## zk-SNARK
+
+**Z**ero-**K**nowledge **S**uccinct **N**on-interactive **Ar**gument of **K**nowledge
+
+- **Zero knowledge** - the proof reveals nothing about the witness that was not revealed by the statement itself.
+- **Succinct** - the proof is small
+- **Non-interactie** - Does not  require live interaction between prover and the verifier.
+- **Proof of knowledge** - if you can compute correct proofs of a statement, you should be able to compute a witness for it.
+
+---
+
+## What can we show?
+
+- NP relation: `function(statement, witness) -> bool`
+
+  - They want to show that they know it (_a proof of knowledge_)
+
+  - ... Without revealing anything about the witness (_ZK_)
+
+- With a small proof even if the witness is large (_succinctness_)
+
+---
+
+## What can we show?
+
+- There are many SNARK schemes to produce succinct ZK proofs of knowledge (_ZK-SNARKs_) for every NP relation. We concentrate on PLONK in this course.
```

Comment:
Just be ready to explain NP in a succinct way :)

No pun intended there.

Reply:
I thnk all mentions of NP are inherited from @AlistairStewart 's slide. I could remove them if there is a consensus on that.

Reply:
I think it is fine to keep in. I am just leaving a note that you should not assume everyone is familiar is all.